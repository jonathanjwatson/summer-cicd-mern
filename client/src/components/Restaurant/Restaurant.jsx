import React from "react";
import PropTypes from "prop-types";

const Restaurant = ({ name, address, phone, cuisine }) => {
  return (
    <div className="col-sm-4">
      <h2 className="text-center">{name}</h2>
      <p>{address}</p>
      <p>{phone}</p>
      <p>{cuisine}</p>
    </div>
  );
};

Restaurant.propTypes = {
  name: PropTypes.string,
  address: PropTypes.string,
  phone: PropTypes.string,
  cuisine: PropTypes.string,
};

export default Restaurant;
