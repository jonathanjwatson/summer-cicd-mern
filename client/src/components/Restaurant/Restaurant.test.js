import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Restaurant from "./Restaurant";

afterEach(cleanup);

const props = {
  name: "Olive Garden",
  address: "1313 Mockingbird Lane",
  phone: "888-888-8888",
  cuisine: "Italian",
};

describe("Restaurant", () => {
  it("renders correctly", () => {
    const { getByText } = render(<Restaurant {...props} />);
    expect(getByText(props.name)).toBeTruthy();
    expect(getByText(props.address)).toBeTruthy();
    expect(getByText(props.phone)).toBeTruthy();
    expect(getByText(props.cuisine)).toBeTruthy();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Restaurant {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
