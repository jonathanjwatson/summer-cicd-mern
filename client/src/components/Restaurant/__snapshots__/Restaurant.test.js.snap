// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurant matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-sm-4"
  >
    <h2
      class="text-center"
    >
      Olive Garden
    </h2>
    <p>
      1313 Mockingbird Lane
    </p>
    <p>
      888-888-8888
    </p>
    <p>
      Italian
    </p>
  </div>
</DocumentFragment>
`;
